pinot-spi/src/main/java/org/apache/pinot/spi/queryeventlistener/BrokerQueryEventInfo.java

```diff
@@ -0,0 +1,462 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.queryeventlistener;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.apache.pinot.spi.trace.RequestContext;
+
+
+public class BrokerQueryEventInfo {
+  private String _requestId;
+  private String _brokerId;
+  private String _query;
+  private String _queryStatus;
+  private String _failureJson;
+  private int _errorCode;
+
+  private long _requestArrivalTimeMillis;
+  private int _numServersQueried;
+  private int _numServersResponded = 0;
```

Comment:
nit: all int/long will anyways be initialized to 0. Strings would be null though. Should we initialize those to `""`?

Reply:
I initialised all int and long to 0 but errorCode to -1 just to be on safe side as 0 might be taken as success. Rest all strings are now assigned as "".